File: frontend/src/microservices/hotel/objects/HotelSearchPlannerRequest.ts
export type HotelSearchPlannerRequest = {
  location?: string
  checkInDate?: string
  checkOutDate?: string
}

export const hotelSearchPlannerRequestFromJson = (json: string): HotelSearchPlannerRequest =>
  JSON.parse(json) as HotelSearchPlannerRequest

export const hotelSearchPlannerRequestToJson = (value: HotelSearchPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/HotelSearchPlannerRequest.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class HotelSearchPlannerRequest(
    location: Option[String],
    checkInDate: Option[String],
    checkOutDate: Option[String]
)

object HotelSearchPlannerRequest:
  given sourceEncoder: Encoder[HotelSearchPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[HotelSearchPlannerRequest] = deriveDecoder
